pkg/controller/daemon/daemon_controller.go

```diff
@@ -761,6 +761,12 @@ func (dsc *DaemonSetsController) getNodesToDaemonPods(ctx context.Context, ds *a
 	nodeToDaemonPods := make(map[string][]*v1.Pod)
 	logger := klog.FromContext(ctx)
 	for _, pod := range claimedPods {
+		if podutil.IsPodTerminal(pod) && pod.DeletionTimestamp != nil {
```

Comment:
I'm not sure what are the alternatives you are suggesting.

A pod with a terminal phase (Succeeded or Failed) is not consuming any resources and is guranteed not to be restarted by kubelet anymore. It's as if the Pod no longer exists.

What could be the effect of keeping the label?

Reply:
If the pod cannot be removed from etcd store for any reason, then the user can manually rollback to the old revision and resolve the finalizers and the semantics behind it. The finalizer might be set by other component(not owned by the user) and can signal holding of external resources. 

It is a very specific scenario, but It might be safer to keep the revision in case somebody needs it. Once the problematic pod is removed, the controller revision will be removed.

Reply:
Added a boolean to the function so that these pods can be considered in the history.